sim_instr/src/sim_instr_node.cpp

```diff
@@ -56,15 +63,29 @@ int main(int argc, char** argv) {
 
     std::vector<TimeAndCmdVel> lines(std::istream_iterator<TimeAndCmdVel>(file), {});
     std::vector<ros::Timer> timers;
-
-    auto callback = [&pub, &lines](auto) {
-    pub.publish(lines.back().cmd_vel);
-    lines.pop_back();
-    };
-
-    std::transform(lines.cbegin(), lines.cend(), std::back_inserter(timers), [&nh, &callback](const TimeAndCmdVel &directive) { 
-        return nh.createTimer(directive.time, callback, true);
-    });
+	timers.reserve(lines.size());
+
+	bool lines_sorted;
+	for(int i = 0; i < (lines.size() - 1); ++i) {
+		if(!(lines.at(i + 1).time > lines.at(i).time)) {
+			ROS_WARN_STREAM("Commands in the Input file should be in time-increasing order, resolved internally");
+			lines_sorted = false;
+			break;
+		}
+	}
+
+
+	while (!ros::Time::isValid()) {
+		std::cout << "spin once" << std::endl;
+		ros::spinOnce();
+		
+	}
+
+	for (const TimeAndCmdVel &directive : lines) {
```

Comment:
replace this for loop with a call to [`std::transform`](https://en.cppreference.com/w/cpp/algorithm/transform) into a [`std::back_insert_iterator`](https://en.cppreference.com/w/cpp/iterator/back_insert_iterator) created using [`std::back_inserter`](https://en.cppreference.com/w/cpp/iterator/back_inserter).